streams/runners/storm/sampling/src/main/java/com/hortonworks/streamline/streams/sampling/service/storm/StormTopologySamplingService.java

```diff
@@ -0,0 +1,257 @@
+            if (!StringUtils.isEmpty(lines)) {
+                addSampledEvents(lines, componentId, events);
+            }
+            match = PREV_START_REGEX.matcher(res);
+            prevOffset = match.find() ? Long.parseLong(match.group(1)) : null;
+            match = NEXT_START_REGEX.matcher(res);
+            nextOffset = match.find() ? Long.parseLong(match.group(1)) : null;
+            cur = qps.desc() ? prevOffset : nextOffset;
+        } while (events.size() < qps.count() && cur != null);
+
+        long next = 0;
+        Integer length = null;
+        if (!events.isEmpty()) {
+            if (events.size() > qps.count()) {
+                int i = qps.desc() ? events.size() - qps.count() : 0;
+                int j = qps.desc() ? events.size() : qps.count();
+                events = events.subList(i, j);
+            }
+            if (qps.desc()) {
+                if (qps.start() != null) {
+                    next = qps.start() + events.get(0).getStartOffset() - qps.length() + 1;
+                } else if (prevOffset != null) {
+                    next = prevOffset + events.get(0).getStartOffset() + 1;
+                }
+                if (next >= 0) {
+                    length = qps.length();
+                } else if (qps.start() > 0 && qps.start() < qps.length()){
+                    length = events.get(0).getStartOffset() + qps.start().intValue() + 1;
+                } else {
+                    length = events.get(0).getStartOffset() + 1;
+                }
+                next = Math.max(next, 0);
+                Collections.reverse(events);
+            } else {
+                SampledEvent lastEvent = events.get(events.size() - 1);
+                next = (qps.start() == null ? 0 : qps.start()) + lastEvent.getStartOffset() + lastEvent.getLength() - 1;
+                length = qps.length() == null ? Integer.valueOf(BYTES_TO_FETCH) : qps.length();
+            }
+        }
+        return buildSampledEvents(next, length, events);
+    }
+
+    private SampledEvents buildSampledEvents(Long next, Integer length, List<SampledEvent> events) {
+        return new SampledEvents() {
+            @Override
+            public Collection<SampledEvent> getEvents() {
+                return events;
+            }
+
+            @Override
+            public Long getNext() {
+                return next;
+            }
+
+            @Override
+            public Integer getLength() {
+                return length;
+            }
+        };
+    }
+
+    private void addSampledEvents(String lines, String queriedComponentId, List<SampledEvent> events) {
+        // Timestamp, Component name, Component task-id, MessageId (incase of anchoring), List of emitted values
+        int offset = 0;
+        for (String line : lines.split("\n")) {
+            line = StringEscapeUtils.unescapeHtml(line);
+            int componentStart = line.indexOf(',') + 1;
+            if (componentStart != -1) {
```

Comment:
Given that you add 1 after `indexOf`, componentStart is in worst case `0`, not `-1`

Reply:
thanks for noticing.